tests/agraph_backend_tests.cpp

```diff
@@ -181,4 +181,21 @@ TEST_F(AGraphBackend, get_utilized_commands) {
   }
   ASSERT_EQ(num_used_commands, 8);
 }
+
+TEST_F(AGraphBackend, reduce_stack) {
+  Eigen::ArrayX3i expected_stack(8, 3);
+  expected_stack << 0, 0, 0,
+                    0, 1, 1,
+                    1, 0, 0,
+                    1, 1, 1,
+                    5, 3, 1,
+                    2, 4, 2,
+                    4, 5, 0,
+                    3, 6, 0;  // (C_1 / X_1 + C_0) * X_0 - X_0
+
+  Eigen::ArrayX3i reduced_stack = ReduceStack(simple_stack);
+
+  ASSERT_TRUE(reduced_stack.isApprox(expected_stack));
+}
+
```

Comment:
I wanted to test `simplify_stack` using mocking, but it seems like it would be pretty gnarly for [normal methods](https://google.github.io/googletest/gmock_cook_book.html#mocking-free-functions) or impossible for [static methods](https://google.github.io/googletest/gmock_cook_book.html#mocking-free-functions) without redesigning the code for test cases.

I figure testing the simplification on the Python side using the C++ backend is sufficient for now until we implement the simplification in C++.